Implement @Attribute Injection. (#5547)
line/armeria: site/src/pages/docs/server-annotated-service.mdx

}
```

> ### Getting an AttributeKey

Reviewer: Getting a custom attribute value

Armeria also provides the <type://@Attribute> annotation to inject an <type://AttributeKey> value from
<type://RequestContext> into a parameter. The parameter annotated with <type://@Attribute> can be compatible
with all types corresponding to the <type://AttributeKey> stored within the <type://RequestContext>. The
<type://@Attribute> annotation has two parameters: `name` and `prefix`, as shown in the example below.
The parameter `name` corresponds to the <type://AttributeKey>'s name, and `prefix` corresponds to the type of
the <type://AttributeKey>. In short, it's `AttributeKey.valueOf(prefix, name)`. Also, the parameter `prefix` is
optional. If the `prefix` is not declared in <type://@Attribute>, `AnnotatedService Type` and `DefaultType` are
used to retrieve the value from <type://RequestContext>. In the example below, `MyAnnotatedService` is used as

> the `ServiceType`.

Reviewer: ```
Armeria also provides the <type://@Attribute> annotation to inject a
custom attribute value of a <type://RequestContext> into the annotated
parameter. The <type://@Attribute> annotation has two attributes:
`value` and  `prefix`, as shown in the example below. If both `prefix` 
and `value` are specified, `AttributeKey.valueOf(prefix, value)` will
be used to look up the `AttributeKey` of the custom attribute.
If `prefix` is unspecified, the class of the annotated service will be
tried as the `prefix` (e.g. `Attribute.valueOf(MyAnnotatedService.class, value)`),
and then if there's no such attribute, `Attribute.valueOf(value)`
will be tried as a fallback.
```

```java

> public class MyAnnotatedService {
> 
>     @Get("/hello1")
>     public HttpResponse hello1(@Attribute(name = "Name", prefix = OtherService.class) String name) { ... }
> 
>     @Post("/hello2")
>     public HttpResponse hello2(@Attribute(age = "Age") int age) { ... }
> 
> }

Reviewer: ```suggestion
public class MyAttributes {
    public static final AttributeKey<String> USERNAME =
            AttributeKey.valueOf(MyAttributes.class, "USERNAME");
}

public class MyAnnotatedService {

    // Uses `AttributeKey.valueOf(MyAttributes.class, "USERNAME")`
    @Get("/hello1")
    public HttpResponse hello1(@Attribute(prefix = MyAttributes.class,
                                          value = "USERNAME")
                               String username) { ... }

    // Uses `AttributeKey.valueOf(MyAnnotatedService.class, "USERNAME")`
    // if possible. If non-existent, use `AttributeKey.valueOf("USERNAME")`.
    @Post("/hello2")
    public HttpResponse hello2(@Attribute("USERNAME") String username) { ... }
}
```

```

> Note that if there is no value from <type://RequestContext> to inject, `IllegalArgumentException` will occur at runtime.

Reviewer: ```suggestion
Note that if no attribute for the specified `AttributeKey` is found in <type://RequestContext>, `IllegalArgumentException` will occur at runtime.
```
